modules/core/src/test/java/org/apache/ignite/internal/processors/database/BPlusTreeSelfTest.java

```diff
@@ -1744,16 +1756,15 @@ public void doTestSizeForRandomPutRmvMultithreadedAsync(final int rmvPutSlidingW
      * the method is picking up new pages which are concurrently added to the tree until the new pages are not added
      * anymore. Test verifies that despite livelock condition a size from a valid range is returned.
      *
-     * NB: This test has to be changed with the integration of IGNITE-3478.
-     *
      * @throws Exception if test failed
      */
     @Test
     public void testPutSizeLivelock() throws Exception {
         MAX_PER_PAGE = 5;
         CNT = 800;
 
-        final int SLIDING_WINDOW_SIZE = 16;
+        // Sliding window size should be grater than cores to avoid races between puts and removes values in the tree.
```

Comment:
```suggestion
        // Sliding window size should be greater than the amount of CPU cores to avoid races between puts and removes in the tree.
```

Reply:
Fix it